File: Problems/0401-binary-watch.py
class Solution(object):
    def readBinaryWatch(self, turnedOn):
        """
        :type turnedOn: int
        :rtype: List[str]
        """
        ans = []

        for h in range(12):
            hc = bin(h).count('1')
            for m in range(60):
                if hc + bin(m).count('1') == turnedOn:
                    ans.append("%d:%02d" % (h, m))

        return ans

File: Problems/0401-binary-watch.cpp
#include <bits/stdc++.h>
using namespace std;

// LeetCode solution starts from here
class Solution {
public:
    vector<string> readBinaryWatch(int turnedOn) {
        vector<string> ans;

        for (int h = 0; h < 12; h++) {
            int hc = __builtin_popcount(h);

            for (int m = 0; m < 60; m++) {
                if (hc + __builtin_popcount(m) == turnedOn) {
                    string time = to_string(h) + ":";
                    if (m < 10)
                        time += "0";
                    time += to_string(m);
                    ans.push_back(time);
                }
            }
        }

        return ans;
    }
};